### PR TITLE
[NFC, sanitizer_symbolizer] Split Fuchsia and Markup.

### PR DESCRIPTION
**Pull request by avillega**
This PR separates parts of the symbolizer markup
implementation that are Fuchsia OS specific. This
is in preparation of enabling symbolizer markup
in other OSs.

### CODE REVIEW

File: compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+//===----------------------------------------------------------------------===//
+
+#include "sanitizer_platform.h"
+#if SANITIZER_FUCHSIA
+
+#  include <limits.h>
+#  include <unwind.h>
+
+#  include "sanitizer_common.h"
+#  include "sanitizer_stacktrace.h"
```
Comment (PiJoules) [inline compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp:21]: nit: remove the whitespace between `#` and `include`

File: compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace __sanitizer {
+
+#  if SANITIZER_CAN_SLOW_UNWIND
```
Comment (PiJoules) [inline compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp:25]: same here

File: compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  PopStackFrames(Min(to_pop, static_cast<uptr>(1)));
+  trace_buffer[0] = pc;
+}
+
+void BufferedStackTrace::UnwindSlow(uptr pc, void *context, u32 max_depth) {
+  CHECK(context);
+  CHECK_GE(max_depth, 2);
+  UNREACHABLE("signal context doesn't exist");
+}
+#  endif  //  SANITIZER_CAN_SLOW_UNWIND
```
Comment (PiJoules) [inline compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp:62]: same here

**Review by PiJoules — APPROVED** [review]
LGTM

File: compiler-rt/CMakeLists.txt [hunk 4]
```diff
@@ -337,6 +337,7 @@ if(NOT COMPILER_RT_HAS_FVISIBILITY_HIDDEN_FLAG)
   append_list_if(COMPILER_RT_HAS_FVISIBILITY_INLINES_HIDDEN_FLAG -fvisibility-inlines-hidden SANITIZER_COMMON_CFLAGS)
 endif()
 append_list_if(COMPILER_RT_HAS_FNO_LTO_FLAG -fno-lto SANITIZER_COMMON_CFLAGS)
+append_list_if(COMPILER_RT_ENABLE_SYMBOLIZER_MARKUP -DENABLE_SANITIZER_SYMBOLIZER_MARKUP SANITIZER_COMMON_CFLAGS)
```
[the remark this answers is not in the document]
Comment (avillega) [reply compiler-rt/CMakeLists.txt]: This change was a cherry pick from a local change and this was not intended. I had removed it.

File: compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp (hunk 1 above)
Comment (avillega) [reply compiler-rt/lib/sanitizer_common/sanitizer_unwind_fuchsia.cpp:21]: These were added by git-clang-format, if I remove them the check for code formatting will fail.

**Comment by vitalybuka** [comment]
lgtm

**Pull request merged**